File: app/src/main/java/mx/mobile/solution/nabia04/room_database/DataConverter.java
package mx.mobile.solution.nabia04.room_database;


import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;
import java.util.Map;

import androidx.room.TypeConverter;


public class DataConverter {
    @TypeConverter
    public String getStrPayments(String[] payments) {
        if (payments == null) {
            return (null);
        }
        Gson gson = new Gson();
        Type type = new TypeToken<String[]>() {}.getType();
        return gson.toJson(payments, type);
    }

    @TypeConverter
    public String[] getPaymentFrmStr(String payments) {

        if (payments == null) {
            return (null);
        }
        Gson gson = new Gson();
        Type type = new TypeToken<String[]>() {
        }.getType();
        return gson.fromJson(payments, type);
    }

    public String getPaymentToString(List<Map<String, String>> countryLang) {
        if (countryLang == null) {
            return (null);
        }
        Gson gson = new Gson();
        Type type = new TypeToken<List<Map<String, String>>>() {
        }.getType();
        return gson.toJson(countryLang, type);
    }

    @TypeConverter
    public List<Map<String, String>> getPaymentList(String countryLangString) {
        if (countryLangString == null) {
            return (null);
        }
        Gson gson = new Gson();
        Type type = new TypeToken<List<Map<String, String>>>() {
        }.getType();
        return gson.fromJson(countryLangString, type);
    }


}

File: nabia04_beta1/src/main/java/mx/mobile/solution/nabia04_beta1/data/converters/DataConverter.kt
package mx.mobile.solution.nabia04_beta1.data.converters

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class DataConverter {
    @TypeConverter
    fun getStrPayments(payments: Array<String?>?): String? {
        if (payments == null) {
            return null
        }
        val gson = Gson()
        val type = object : TypeToken<Array<String?>?>() {}.type
        return gson.toJson(payments, type)
    }

    @TypeConverter
    fun getPaymentFrmStr(payments: String?): Array<String>? {
        if (payments == null) {
            return null
        }
        val gson = Gson()
        val type = object : TypeToken<Array<String?>?>() {}.type
        return gson.fromJson(payments, type)
    }

    fun getPaymentToString(countryLang: List<Map<String?, String?>?>?): String? {
        if (countryLang == null) {
            return null
        }
        val gson = Gson()
        val type = object : TypeToken<List<Map<String?, String?>?>?>() {}.type
        return gson.toJson(countryLang, type)
    }

    @TypeConverter
    fun getPaymentList(countryLangString: String?): List<Map<String, String>>? {
        if (countryLangString == null) {
            return null
        }
        val gson = Gson()
        val type = object : TypeToken<List<Map<String?, String?>?>?>() {}.type
        return gson.fromJson(countryLangString, type)
    }
}